### PR TITLE
[P2P] chore: remove unnecessary `stdnetwork` package

### LINKED ISSUE #266: [TECHDEBT] [P2P] Remove stdNetwork and simplify `Network` interface
## Objective

Tend for the TODO items:

```
./p2p/types/network.go:// TECHDEBT(olshansky): When we delete `stdnetwork` and only go with `raintree`, this interface can be simplified greatly.
./p2p/stdnetwork/network.go:// TECHDEBT(olshansky): Delete this once we are fully comfortable with RainTree moving forward.
```

Refine the goals / priority of this issue since it looks like it has dependencies like "being comfortable with RainTree"

## Origin Document

[ Why? Issue justification and/or link to another document]

## Goals

- _\_*REPLACE_ME*: List of things that are high level ideas or goals driving the task_
- [ ] ...

## Deliverable

- _\_*REPLACE_ME*: List of things that are concrete deliverables_
- [ ] ...

## Non-goals / Non-deliverables

- _\_*REPLACE_ME*: List of things that are out of scope_
- ...

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- \__REPLACE_ME_: Make sure to update the testing methodology appropriately\_
- **Task specific tests**: `make ...`
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @deblasis 
**Co-Owners**: @Olshansk

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Description

Removes the unneeded `stdnetwork` package.

## Issue

Covers removal goal of #266. The simplification goal is tracked by #553.
TLDR; should close #266.

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Removed unneeded `stdnetwork` package
- Removed unneeded `use_rain_tree` P2P config field

## Testing

- [ ] `make develop_test`; if any code changes were made
- [x] [Docker Compose LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md); if any major functionality was changed or introduced
- [x] [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any infrastructure or configuration changes were made

## Required Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added, or updated, [`godoc` format comments](https://go.dev/blog/godoc) on touched members (see: [tip.golang.org/doc/comment](https://tip.golang.org/doc/comment))
- [ ] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

**Review by Olshansk — APPROVED** [review]
🧹🧹🧹
